test/runner/configuration/include_test.dart

```diff
@@ -27,4 +26,64 @@ void main() {
     expect(config.filename.pattern, 'test_*.dart');
     expect(config.concurrency, 3);
   });
+
+  test('should merge fields with a base configuration', () async {
+    await d.dir('repo', [
+      d.file('dart_test_base.yaml', r'''
+        tags:
+          hello:
+      '''),
+      d.dir('pkg', [
+        d.file('dart_test.yaml', r'''
+          include: ../dart_test_base.yaml
+          tags:
+            world:
+        '''),
+      ]),
+    ]).create();
+    var path = p.join(d.sandbox, 'repo', 'pkg', 'dart_test.yaml');
+    var config = new Configuration.load(path);
+    expect(config.knownTags, ['hello', 'world']);
+  });
+
+  test('should allow an included file to include a file', () async {
+    await d.dir('repo', [
+      d.file('dart_test_base_base.yaml', r'''
+        tags:
+          tag:
+      '''),
+      d.file('dart_test_base.yaml', r'''
+        include: dart_test_base_base.yaml
+        filename: "test_*.dart"
+      '''),
+      d.dir('pkg', [
+        d.file('dart_test.yaml', r'''
+          include: ../dart_test_base.yaml
+          concurrency: 3
+        '''),
+      ]),
+    ]).create();
+    var path = p.join(d.sandbox, 'repo', 'pkg', 'dart_test.yaml');
+    var config = new Configuration.load(path);
+    expect(config.knownTags, ['tag']);
+    expect(config.filename.pattern, 'test_*.dart');
+    expect(config.concurrency, 3);
+  });
+
+  test('should handle a non-string include field value gracefully', () async {
+    await d.dir('repo', [
+      d.dir('pkg', [
+        d.file('dart_test.yaml', r'''
+          include: 3 # Oops!
+        '''),
+      ]),
+    ]).create();
+    var path = p.join(d.sandbox, 'repo', 'pkg', 'dart_test.yaml');
+    expect(() => new Configuration.load(path), throwsFormatException);
+  });
+
+  test('should not allow an include field in a test config context', () async {
+    expect(() => new Configuration.parse(['--include=dart_test.yaml']),
```

Comment:
I meant like:

```yaml
tags:
  foo:
    include: ../dart_test.yaml
```

Come to think of it, we should also test that:

```yaml
presets:
  foo:
    include: preset.yaml
```

works as expected.

Reply:
Done, thanks!